packages/remme-public-key-storage/src/index.ts

```diff
@@ -193,25 +206,149 @@ class RemmePublicKeyStorage implements IRemmePublicKeyStorage {
+     *
+     * storeResponse.connectToWebSocket((err: Error, res: any) => {
+     *      if (err) {
+     *          console.log(err);
+     *          return;
+     *      }
+     *      console.log(res);
+     * })
+     * ```
+     * @param {Uint8Array} data
+     */
+    public async store(data: Uint8Array): Promise<IBaseTransactionResponse> {
+        let message: NewPubKeyPayload | NewPubKeyStoreAndPayPayload;
+        let payload: Uint8Array;
+        let pubKeyAddress: string;
 
-        const pubKeyAddress = keys.address;
+        message = NewPubKeyStoreAndPayPayload.decode(data).pubKeyPayload.entityHash.byteLength
+            ? NewPubKeyStoreAndPayPayload.decode(data)
+            : NewPubKeyPayload.decode(data);
+
+        if (message instanceof NewPubKeyPayload) {
+            const { entityHash, entityHashSignature } = message;
+
+            checkSha(entityHash.toString());
+
+            const keys = await this._constructKeysFromPayload(message);
+
+            if (!keys.verify(entityHash.toString(), bytesToHex(entityHashSignature))) {
+                throw new Error("Signature not valid");
+            }
+
+            pubKeyAddress = keys.address;
+            payload = data;
+        } else if (message instanceof NewPubKeyStoreAndPayPayload) {
+            const { ownerPublicKey, signatureByOwner, pubKeyPayload } = message;
+            const signedPayload = NewPubKeyPayload.encode(pubKeyPayload).finish();
+
+            const accountKey = await RemmeKeys.construct({
+                keyType: KeyType.ECDSA,
+                publicKey: ownerPublicKey,
+            });
+
+            if (!accountKey.verify(signedPayload, bytesToHex(signatureByOwner))) {
+                throw new Error("Owner signature not valid");
+            }
+
+            const { entityHash, entityHashSignature} = pubKeyPayload;
+            const keys = await this._constructKeysFromPayload(pubKeyPayload);
+
+            if (!keys.verify(entityHash.toString(), bytesToHex(entityHashSignature))) {
+                throw new Error("Signature not valid");
+            }
+            pubKeyAddress = keys.address;
+            payload = NewPubKeyPayload.encode(pubKeyPayload).finish();
+        } else {
+            throw new Error("Invalid payload");
+        }
 
-        const storageSettingsAddress = generateSettingsAddress("remme.settings.storage_pub_key");
         const settingAddress = generateSettingsAddress("remme.economy_enabled");
 
-        const payloadBytes = this._generateTransactionPayload(PubKeyMethod.Method.STORE, payload);
-        const inputs = [
-            pubKeyAddress,
-            storageSettingsAddress,
-            settingAddress,
-            this._zeroAddress,
-        ];
-        const outputs = [
+        const inputsOutputs = [
             pubKeyAddress,
+            this._remmeAccount.address,
             this._zeroAddress,
+            settingAddress,
         ];
```

Comment:
invalid inputsOutputs.
this._remmeAccount.address automaticaly added in remme-transaction-service.
owner_pubkey_address is not set